python/cudf/cudf/core/reshape.py

```diff
@@ -972,10 +992,16 @@ def unstack(df, level, fill_value=None):
             return df
     df = df.copy(deep=False)
     if not isinstance(df.index, cudf.MultiIndex):
-        raise NotImplementedError(
-            "Calling unstack() on a DataFrame without a MultiIndex "
-            "is not supported"
-        )
+        if isinstance(df, cudf.DataFrame):
+            res = df.T.stack(dropna=False)
```

Comment:
Does this pass the typecasting behavior off to transpose? Should we check the dtypes and possibly error here?

Reply:
It seems like both `transpose.pyx` and `libcudf::transpose.cu` checks whether all columns have the same datatype. A clear exception gets raised if the columns are of different types.

Reply:
I would support checking here - imagining what happens here from the user perspective, if I get an error trying to `unstack` a cuDF dataframe, I might wonder why the transpose code is unhappy. 

In general, I think we try and avoid letting libcudf itself serve an error to the user and favor a more surface level python error, usually when I've managed to actually manifest a libcudf error from the python API it means something is very wrong.